client/packages/invoices/src/Returns/DetailView/DetailView.tsx

```diff
@@ -32,34 +32,39 @@ export const DetailView: FC = () => {
   const { data, isLoading } = useReturns.document.outboundReturn();
   const t = useTranslation('distribution');
   const navigate = useNavigate();
-  // const onRowClick = useCallback(
+
+  const onRowClick = () => {};
+
+  // useCallback(
   //   (item: StockOutLineFragment | StockOutItem) => {
   //     onOpen({ item: toItemRow(item) });
   //   },
-  //   [toItemRow, onOpen]
+  //   [toItemRow]
   // );
-  // const onAddItem = (draft?: Draft) => {
+  const onAddItem = () => {};
+  //  (draft?: Draft) => {
   //   onOpen(draft);
   //   setMode(ModalMode.Create);
   // };
 
   if (isLoading) return <DetailViewSkeleton hasGroupBy={true} hasHold={true} />;
 
-  // const tabs = [
-  //   {
-  //     Component: (
-  //       <ContentArea
-  //         // onRowClick={!isDisabled ? onRowClick : null}
-  //         onAddItem={onAddItem}
-  //       />
-  //     ),
-  //     value: 'Details',
-  //   },
-  //   {
-  //     Component: <ActivityLogList recordId={data?.id ?? ''} />,
-  //     value: 'Log',
-  //   },
-  // ];
+  const tabs = [
+    {
+      Component: (
+        <ContentArea
+          onRowClick={onRowClick}
+          onAddItem={onAddItem}
+          rows={data?.lines?.nodes}
```

Comment:
This is not how it's done elsewhere. In Outbound Shipments, rows is fetched in `ContentArea` using an additional hook `useOutbound.line.rows`, but I don't really get what that hook is doing (it seems to be just re-packaging the document query in a different way, but it's really confusing). Hence I'm just passing the columns in for now.  Suggestions welcome.

Reply:
I think how you have it is great to get the table in for now. 

At some point we'll need to bring in grouping by item, and then I think the rows hook will be helpful. It abstracts away the logic to return all the lines, or the item ones, which would really bloat the component if done there.